shenyu-common/src/main/java/org/apache/shenyu/common/utils/SignUtils.java

```diff
@@ -49,16 +52,20 @@ public static SignUtils getInstance() {
      * acquired sign.
      *
      * @param signKey sign key
-     * @param params  params
+     * @param jsonParams json params
+     * @param queryParams  url query params
      * @return sign
      */
-    public static String generateSign(final String signKey, final Map<String, String> params) {
-        final String sign = params.keySet().stream()
+    public static String generateSign(final String signKey, final Map<String, String> jsonParams, final Map<String, String> queryParams) {
+        final String jsonSign = Optional.ofNullable(jsonParams).orElse(EMPTY_HASH_MAP).keySet().stream()
+                .sorted(Comparator.naturalOrder())
+                .map(key -> String.join("", key, jsonParams.get(key)))
```

Comment:
refactor this code

Reply:
How about removing the transStringMap method and refactoring the generateSign method as follows 
```java
public static String generateSign(final String signKey, final Map<String, Object> jsonParams, final Map<String, String> queryParams) {
        final String jsonSign = jsonParams==null?null:jsonParams.keySet().stream()
                .sorted(Comparator.naturalOrder())
                .map(key -> String.join("", key, Objects.toString(jsonParams.get(key), null)))
                .collect(Collectors.joining()).trim();
        final String querySign = queryParams==null?null:queryParams.keySet().stream()
                .sorted(Comparator.naturalOrder())
                .map(key -> String.join("", key, queryParams.get(key)))
                .collect(Collectors.joining()).trim();
        final String sign = String.join("", jsonSign, querySign, signKey);
        // TODO this is a risk for error charset coding with getBytes
        return DigestUtils.md5DigestAsHex(sign.getBytes()).toUpperCase();
    }
```